openerp/addons/base/ir/ir_ui_menu.py

```diff
@@ -337,8 +338,8 @@ def load_menus_root(self, cr, uid, context=None):
 
 
     @api.cr_uid_context
-    @tools.ormcache_context('uid', keys=('lang',))
-    def load_menus(self, cr, uid, context=None):
+    @tools.ormcache_context('uid', 'debug', keys=('lang',))
+    def load_menus(self, cr, uid, debug=False, context=None):
```

Comment:
load_menus should load from cache in both case debug is on/off , in both cases it must load relevant menu i mean if debug mode is off and menu have a technical group, menu should not visible untill user have a technical rights + debug mode is on. 
ok i add new parameter at the end.